src/libraries/illink-sharedframework.targets

```diff
@@ -1,48 +1,60 @@
 <Project>
 
-  <Target Name="ILLinkTrimSharedFramework"
-          AfterTargets="Build"
+  <Target Name="GetSharedFrameworkAssembliesToTrim"
           DependsOnTargets="PrepareForAssembliesTrim">
 
-    <Message Text="Trimming $(PackageRID) runtime pack assemblies with ILLinker..." Importance="high" />
-
     <PropertyGroup>
-      <LibrariesTrimmedArtifactsPath>$([MSBuild]::NormalizePath('$(ILLinkTrimAssemblyArtifactsRootDir)', 'trimmed-runtimepack'))</LibrariesTrimmedArtifactsPath>
+      <SharedFrameworkAssembliesTrimmedArtifactsPath>$([MSBuild]::NormalizePath('$(ILLinkTrimAssemblyArtifactsRootDir)', 'trimmed-runtimepack'))</SharedFrameworkAssembliesTrimmedArtifactsPath>
     </PropertyGroup>
 
+    <ItemGroup>
+      <!-- Include suppression XML files bin-placed in earlier per-library linker run. -->
+      <SharedFrameworkSuppressionsXml Include="$(ILLinkTrimAssemblyRuntimePackSuppressionsXmlsDir)*.xml" />
+      <!-- Collect CoreLib suppression XML files not bin-placed in earlier per-library linker run. CoreLib doesn't use bin-place logic. -->
+      <SharedFrameworkSuppressionsXml Include="$(CoreLibSharedDir)ILLink\ILLink.Suppressions.LibraryBuild.xml" />
+      <SharedFrameworkSuppressionsXml Condition="'$(RuntimeFlavor)' == 'CoreCLR'" Include="$(CoreClrProjectRoot)System.Private.CoreLib\$(ProjectILLinkSuppressionsFile).LibraryBuild.xml" />
+      
+      <SharedFrameworkAssemblyToTrim Include="$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)*.dll" />
+      <SharedFrameworkAssemblyToTrim Include="$(SystemPrivateCoreLibPath)" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <SharedFrameworkAssemblyTrimmed Include="$(SharedFrameworkAssembliesTrimmedArtifactsPath)*.*" />
+    </ItemGroup>
+
+  </Target>
+
+  <Target Name="ILLinkTrimSharedFramework"
+          AfterTargets="Build"
+          DependsOnTargets="GetSharedFrameworkAssembliesToTrim;PrepareForAssembliesTrim"
```

Comment:
Same comment about PrepareForAssembliesTrim probably not being necessary here.

Reply:
Same as above, it's a direct dependency.